tutorials/llm/slimpajama/data_pipeline.ipynb

```diff
@@ -0,0 +1,328 @@
+    "1. Create directories `HOST_DATA_PATH/tokenizer` and `HOST_DATA_PATH/slimpajama`.\n",
+    "1. Download the Llama 3 `tokenizer.model` file either from [Huggingface](https://huggingface.co/meta-llama/Llama-2-7b/blob/main/tokenizer.model) or https://www.llama.com/llama-downloads/ and place it at `{HOST_DATA_PATH}/tokenizer/tokenizer.model`.\n",
+    "    For HF, you can do it by running \n",
+    "    ```bash\n",
+    "    HF_TOKEN=... huggingface-cli download meta-llama/Llama-2-7B tokenizer.model --local-dir {HOST_DATA_PATH}/tokenizer/\n",
+    "    ```\n",
+    "\n",
+    "> [!NOTE]\n",
+    "> All code for this tutorial can be found at https://github.com/NVIDIA/NeMo/tree/main/examples/llm/slimpajama."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "\n",
+    "from data.download import download_slimpajama\n",
+    "from data.extract import run_extraction\n",
+    "from data.preprocess import preprocess_data\n",
+    "\n",
+    "HOST_DATA_PATH = \"/data\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Downloading Data\n",
+    "\n",
+    "First, we will configure the task to download data from Huggingface. We will use the Huggingface CLI for this. The function that configures the download script can be found [here](./data/download.py)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "download_task = download_slimpajama(\n",
+    "    include_pattern='--include \"train/chunk1/*_100*zst\"',\n",
+    ")\n",
+    "\n",
+    "# The configured script looks like below\n",
+    "print(download_task.inline)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extracting Data\n",
+    "\n",
+    "The downloaded data is in compressed zst format. We need to extract it into jsonl files. For that, we will configure the `extract_data` function defined [here](./data/extract.py). This function also allows excluding certain sources. By default, we exclude all data from `RedPajamaBook` set, but this setting is configurable."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run_extraction??"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "extract_task = run.Partial(run_extraction, data_dir=\"/data/slimpajama\")\n",
+    "extract_task"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Concatenating Data\n",
+    "\n",
+    "This optional step concatenates small jsonl files into a single large jsonl files. The example script is [here](./data/concat.sh) but feel free to change it based on your needs."
```

Comment:
fix capitalization and punctuation

This optional step concatenates small JSONL files into a single large JSONL files. The example script is [here](https://github.com/NVIDIA/NeMo/blob/a4f0517574f9327d77d34e3f93a8f61b5ff7cd0a/tutorials/llm/slimpajama/data/concat.sh), but feel free to change it based on your needs.